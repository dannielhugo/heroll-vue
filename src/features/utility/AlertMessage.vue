<template>
  <el-alert v-if="show" :title="message" :type="type" />
</template>

<script setup lang="ts">
import { storeToRefs } from 'pinia';
import { useMessageStore } from '@/stores/message';

const messageStore = useMessageStore();
const { show, message, type } = storeToRefs(messageStore);
</script>

<style scoped></style>
