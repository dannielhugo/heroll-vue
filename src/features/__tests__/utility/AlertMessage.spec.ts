import { mount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';
import ElementPlus from 'element-plus';
import AlertMessage from '@/features/utility/AlertMessage.vue';

describe('AlertMessage', () => {
  test('should mount component', async () => {
    expect(AlertMessage).toBeTruthy();

    const wrapper = mount(AlertMessage, {
      global: {
        plugins: [
          ElementPlus,
          createTestingPinia({
            initialState: {
              message: {
                message: 'An error',
                type: 'error',
                show: true,
              },
            },
          }),
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  test('should mount component and not show', async () => {
    expect(AlertMessage).toBeTruthy();

    const wrapper = mount(AlertMessage, {
      global: {
        plugins: [
          ElementPlus,
          createTestingPinia({
            initialState: {
              message: {
                message: 'An error',
                type: 'error',
                show: false,
              },
            },
          }),
        ],
      },
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
