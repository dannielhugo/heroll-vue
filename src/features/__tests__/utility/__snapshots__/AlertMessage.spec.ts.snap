// Vitest Snapshot v1

exports[`AlertMessage > should mount component 1`] = `
"<transition-stub>
  <div class=\\"el-alert el-alert--error is-light\\" role=\\"alert\\">
    <!--v-if-->
    <div class=\\"el-alert__content\\"><span class=\\"el-alert__title\\">An error</span>
      <!--v-if--><i class=\\"el-icon el-alert__close-btn\\"><svg viewBox=\\"0 0 1024 1024\\" xmlns=\\"http://www.w3.org/2000/svg\\">
          <path fill=\\"currentColor\\" d=\\"M764.288 214.592 512 466.88 259.712 214.592a31.936 31.936 0 0 0-45.12 45.12L466.752 512 214.528 764.224a31.936 31.936 0 1 0 45.12 45.184L512 557.184l252.288 252.288a31.936 31.936 0 0 0 45.12-45.12L557.12 512.064l252.288-252.352a31.936 31.936 0 1 0-45.12-45.184z\\"></path>
        </svg></i>
    </div>
  </div>
</transition-stub>"
`;

exports[`AlertMessage > should mount component and not show 1`] = `"<!--v-if-->"`;
